src/fourier.jl

```diff
@@ -377,34 +391,34 @@ function avg_pds_from_iterable(
             )
         end
 
-        # Accumulate the total sum cross spectrum
-        cross = sum_if_not_none_or_initialize(cross, cs_seg)
-        unnorm_cross = sum_if_not_none_or_initialize(unnorm_cross, unnorm_power)
+        if isnothing(cross)
+            cross = copy(cs_seg)
+        else
+            cross .+= cs_seg
+        end
 
+        if isnothing(unnorm_cross)
+            unnorm_cross = copy(unnorm_power_filtered)
+        else
+            unnorm_cross .+= unnorm_power_filtered
+        end
         n_ave += 1
     end
 
-    # If there were no good intervals, return nothing
     if isnothing(cross)
         return nothing
     end
 
-    # Calculate the mean number of photons per chunk
     n_ph = sum_of_photons / n_ave
-    # Calculate the mean number of photons per bin
     common_mean = n_ph / n_bin
 
     if !isnothing(common_variance)
-        # Note: the variances we summed were means, not sums.
```

Comment:
I don't understand why you have removed all of these comments. Could you explain?